src/lava/magma/compiler/builder.py

```diff
@@ -353,11 +353,14 @@ def build(self):
         """
 
         # Create the ProcessModel
+        pm = self.proc_model()
```

Comment:
Are you using the latest code? I thought I told this Sumit already. 
The ProcModel should simply accept proc_params and model_id as constructor arguments. There is no reason to set these arguments externally by the builder. The code that follows is unnecessary clutter.

The AbstractProcess should just start out with an empty proc_param dictionary that may get populated by the user or not. The compiler should always pass that dictionary to the builder and then to the ProcModel via the respective constructors. 

Much cleaner...